compiler/dalgona/src/Dalgona.cpp

```diff
@@ -114,14 +119,17 @@ void Dalgona::initialize(const std::string &input_model_path)
   // Initialize interpreter
   _interpreter = std::make_unique<luci_interpreter::Interpreter>(_module.get());
 
-  // TODO Attach Python hooks to interpreter
+  _hooks = std::make_unique<PythonHooks>(_interpreter.get());
+
+  _interpreter->attachObserver(_hooks.get());
 }
 
 void Dalgona::runAnalysisWithH5Input(const std::string &input_data_path,
                                      const std::string &analysis_path,
                                      const std::string &analysis_args)
 {
-  // TODO Import analysis code
+  py::object scope = py::module::import("__main__").attr("__dict__");
```

Comment:
I've just seen the pybind document at https://pybind11.readthedocs.io/en/stable/advanced/pycpp/utilities.html#eval.

The example codes look like below.
```cxx
// Evaluate in scope of main module
py::object scope = py::module_::import("__main__").attr("__dict__");
```
Where `module` keyword here has extra underscore.

Is there any difference between `py::module` and `py::module_`?

Reply:
Thanks for careful reading! I didn't even know there was module_.

I've found the history from pybind11 doc.
![image](https://user-images.githubusercontent.com/5449554/191174847-7cbc6e97-be14-4ae2-9a53-0d1a11a88e96.png)

https://github.com/pybind/pybind11/pull/2489

module_ was introduced to avoid conflict with module keyword in cpp. In dalgona, module is used with namespace, so such a conflict will not occur :)